assets/php/class-wgpb-block-library.php

```diff
@@ -342,94 +346,85 @@ public static function register_blocks() {
 	}
 
 	/**
-	 * Output useful globals before printing any script tags.
+	 * Get the schema for the contentVisibility attribute
 	 *
-	 * These are used by @woocommerce/components & the block library to set up defaults
-	 * based on user-controlled settings from WordPress.
+	 * @return array List of block attributes with type and defaults.
+	 */
+	public static function get_schema_content_visibility() {
+		return array(
+			'type'       => 'object',
+			'properties' => array(
+				'title'  => self::get_schema_boolean( true ),
+				'price'  => self::get_schema_boolean( true ),
+				'rating' => self::get_schema_boolean( true ),
+				'button' => self::get_schema_boolean( true ),
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for the orderby attribute.
 	 *
-	 * @since 2.0.0
+	 * @return array Property definition of `orderby` attribute.
 	 */
-	public static function print_script_settings() {
-		global $wp_locale;
-		$code           = get_woocommerce_currency();
-		$product_counts = wp_count_posts( 'product' );
+	public static function get_schema_orderby() {
+		return array(
+			'type'    => 'string',
+			'enum'    => array( 'date', 'popularity', 'price_asc', 'price_desc', 'rating', 'title' ),
+			'default' => 'date',
+		);
+	}
 
-		// NOTE: wcSettings is not used directly, it's only for @woocommerce/components
-		//
-		// Settings and variables can be passed here for access in the app.
-		// Will need `wcAdminAssetUrl` if the ImageAsset component is used.
-		// Will need `dataEndpoints.countries` if Search component is used with 'country' type.
-		// Will need `orderStatuses` if the OrderStatus component is used.
-		// Deliberately excluding: `embedBreadcrumbs`, `trackingEnabled`.
-		$settings = array(
-			'adminUrl'      => admin_url(),
-			'wcAssetUrl'    => plugins_url( 'assets/', WC_PLUGIN_FILE ),
-			'siteLocale'    => esc_attr( get_bloginfo( 'language' ) ),
-			'currency'      => array(
-				'code'      => $code,
-				'precision' => wc_get_price_decimals(),
-				'symbol'    => get_woocommerce_currency_symbol( $code ),
-				'position'  => get_option( 'woocommerce_currency_pos' ),
-			),
-			'stockStatuses' => wc_get_product_stock_status_options(),
-			'siteTitle'     => get_bloginfo( 'name' ),
-			'dataEndpoints' => array(),
-			'l10n'          => array(
-				'userLocale'    => get_user_locale(),
-				'weekdaysShort' => array_values( $wp_locale->weekday_abbrev ),
+	/**
+	 * Get the schema for the alignment property.
+	 *
+	 * @return array Property definition for align.
+	 */
+	public static function get_schema_align() {
+		return array(
+			'type' => 'string',
+			'enum' => array( 'left', 'center', 'right', 'wide', 'full' ),
+		);
+	}
+
+	/**
+	 * Get the schema for a list of IDs.
+	 *
+	 * @return array Property definition for a list of numeric ids.
+	 */
+	public static function get_schema_list_ids() {
```

Comment:
Do these need to be public? If they are only used in this class they can be protected instead. It also makes future changes easier since they won't be depended on elsewhere.

Reply:
I've updated the schema-related functions, though it sounds like more of the functions in this file could also be `protected`, I'm not going to change that here.